nil-docs/zkllvm/overview/what-is-zkllvm.mdx

```diff
@@ -0,0 +1,161 @@
+
+:::
+
+## Pipeline
+
+### Overview
+
+The following diagram outlines the key components and stages of the zkLLVM circuit generation pipeline. Note that some of these components are currently unsupported.
+
+```mermaid
+%%{
+  init: {
+    'theme': 'base',
+    'themeVariables': {
+      'primaryColor': '#0f0f0f',
+      'primaryTextColor': '#f1f1f1',
+      'primaryBorderColor': '#f1f1f1',
+      'lineColor': '#87B6FC',
+      'secondaryColor': '#f1f1f1',
+      'tertiaryColor': '#2f2f2f',
+    },
+    'flowchart':
+    {
+        'defaultRenderer': 'elk',
+        'curve': 'step'
+    }
+  }
+}%%
+flowchart TB
+    classDef subgraphTitleHidden display: none;
+    Initial:::subgraphTitleHidden
+    classDef subgraphTitle margin-right: 650px;
+    classDef subgraphTitleSDK margin-right: 250px;
+    Backend:::subgraphTitle
+    zkLLVM:::subgraphTitle
+    SDK:::subgraphTitleSDK
+    Assigner:::subgraphTitle
+    Final:::subgraphTitleHidden
+    CPP(C++ code)
+    RUST(Rust code)
+    NN(Neural network in ONNX format)
+    RUSTSDK(Rust SDK)
+    CPPSDK(C++ SDK)
+    SQL(SQL statement)
+    CLANGFRONT(Clang frontend)
+    RUSTFRONT(Rust frontend)
+    SQLFRONT(SQL frontend)
+    ONNXFRONT(ONNX frontend)
+    LLVMIR(LLVM IR)
+    R1CS(Blueprint R1CS subtarget)
+    PLONK(Blueprint PLONK subtarget)
+    AIR(Blueprint AIR subtarget)
+    R1CSASS(Blueprint R1CS assigner)
+    PLONKASS(Blueprint PLONK assigner)
+    AIRASS(Blueprint AIR assigner)
+    R1CSPS(R1CS-compatible proof system)
+    PLCPS(Placeholder proof system)
+    AIRPS(AIR-compatible proof system)
+    CPP --> CPPSDK
+    RUST --> RUSTSDK
+    CPPSDK --> CLANGFRONT
+    RUSTSDK --> RUSTFRONT
+    SQL --> SQLFRONT
+    NN --> ONNXFRONT
+    CLANGFRONT --> LLVMIR
+    RUSTFRONT --> LLVMIR
+    SQLFRONT --> LLVMIR
+    ONNXFRONT --> LLVMIR
+    LLVMIR --> R1CS
+    LLVMIR --> PLONK
+    LLVMIR --> AIR
+    R1CS --> R1CSASS
+    PLONK --> PLONKASS
+    AIR --> AIRASS
+    R1CSASS --> R1CSPS
+    AIRASS --> AIRPS
+    R1CSASS --> PLCPS
+    PLONKASS --> PLCPS
+    AIRASS --> PLCPS
+    subgraph Initial
+    CPP
+    RUST
+    SQL
+    NN
+    end
+    subgraph SDK
+    CPPSDK
+    RUSTSDK
+    end
+    subgraph zkLLVM
+    CLANGFRONT
+    RUSTFRONT
+    SQLFRONT
+    ONNXFRONT
+    LLVMIR
+    end
+    subgraph Backend
+    R1CS
+    PLONK
+    AIR
+    end
+    subgraph Assigner
+    R1CSASS
+    PLONKASS
+    AIRASS
+    end
+    subgraph Final
+    R1CSPS
+    PLCPS
+    AIRPS
+    end
+```
+
+### Step-by-step
+
+Working with zkLLVM involves the following steps.
+
+#### Write the code in C++ or Rust
+
+There is no need to use any special syntax when working with zkLLVM. A circuit is defined by simply adding a `[[circuit]]` directive before a function. With some modifications, almost any existing C++ or Rust code can be reused for zkLLVM.
```

Comment:
This is a C++ syntax for attribute. Since we are covering here both C++ and Rust, maybe it makes sense to mention Rust attribute (`#[circuit]`) to avoid confusion.

Reply:
added!